app/uk/gov/hmrc/tai/viewModels/IncomeSourceViewModel.scala

```diff
@@ -116,13 +139,18 @@ object IncomeSourceViewModel extends ViewModelHelper {
   ): IncomeSourceViewModel = {
 
     val endDate: Option[String] = employment.endDate.map(Dates.formatDate(_))
-    val detailsLinkLabel = taxCodeIncome.componentType match {
-      case EmploymentIncome if employment.employmentStatus == Live =>
-        messages("tai.incomeTaxSummary.employmentAndBenefits.link")
-      case EmploymentIncome if employment.employmentStatus != Live => messages("tai.incomeTaxSummary.employment.link")
-      case PensionIncome                                           => messages("tai.incomeTaxSummary.pension.link")
-      case _                                                       => messages("tai.incomeTaxSummary.income.link")
-    }
+
+    def getLinkLabel(messageKey: String): String =
+      taxCodeIncome.componentType match {
+        case EmploymentIncome if employment.employmentStatus == Live =>
```

Comment:
Might be a little more readable rendered like this:
(taxCodeIncome.componentType, employment.employmentStatus) match {
      case (EmploymentIncome, Live) => etc